api/src/main/java/net/wesjd/anvilgui/AnvilGUI.java

```diff
@@ -58,14 +65,20 @@ public class AnvilGUI {
      * A state that decides where the anvil GUI is able to be closed by the user
      */
     private final boolean preventClose;
+
+    /**
+     * This boolean is used to determine if the event should be cancelled or not
+     */
+    private final boolean isInteractive;
+
     /**
      * An {@link Consumer} that is called when the anvil GUI is closed
      */
     private final Consumer<Player> closeListener;
     /**
-     * An {@link BiFunction} that is called when the {@link Slot#OUTPUT} slot has been clicked
+     * An {@link Function5} that is called when the {@link Slot#OUTPUT} slot has been clicked
      */
-    private final BiFunction<Player, String, Response> completeFunction;
+    private final Function5<Player, String, ItemStack, ItemStack, Response> completeFunction;
```

Comment:
For the sake of sanity for the programmer, I think we should go ahead and make the `completeFunction` a simple `Function<Completion, Response>` where a `Completion` would contain all of the data like the player who clicked, etc.

This would be very scalable in the future if we need to make any other changes while avoiding another breaking change.

Reply:
Okay, upload the suggested changes